.github/workflows/test.yml

```diff
@@ -0,0 +1,133 @@
+name: Test
+
+on:
+  # Run on pushes to `master` and on all pull requests.
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - '**.md'
+  pull_request:
+  # Allow manually triggering the workflow.
+  workflow_dispatch:
+
+jobs:
+  #### TEST STAGE ####
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      # Keys:
+      # - experimental: Whether the build is "allowed to fail".
+      matrix:
+        # The GHA matrix works different from Travis.
+        # You can define jobs here and then augment them with extra variables in `include`,
+        # as well as add extra jobs in `include`.
+        # @link https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix
+        #
+        # Note: while WPCS 3.0.0 is under development, the matrix will use `dev-master`.
+        # Once it has been released and YoastCS has been made compatible, the matrix should switch (back)
+        # WPCS `dev-master` to `dev-develop`.
+        php: ['5.4', '5.5', '5.6', '7.0', '7.1', '7.2', '7.3', '7.4', '8.0']
+        phpcs_version: ['3.6.0', 'dev-master']
+        wpcs_version: ['2.3.0', 'dev-master']
+        experimental: [false]
+
+        include:
+          # Experimental builds. These are allowed to fail.
+
+          # PHP nightly
+          - php: '8.1'
+            phpcs_version: 'dev-master'
+            wpcs_version: 'dev-master'
+            experimental: true
+          # Test against WPCS unstable. Re-enable when WPCS is not in dev for the next major.
+          #- php: '8.0'
+          #  phpcs_version: 'dev-master'
+          #  wpcs_version: 'dev-develop'
+          #  experimental: true
+
+          # Test against the next major of PHPCS. Temporarily disabled due to upstream bugs.
+          #- php: '7.4'
+          #  phpcs_version: '4.0.x-dev'
+          #  wpcs_version: 'dev-develop'
+          #  experimental: true
+
+    name: "Test${{ matrix.phpcs_version == 'dev-master' && matrix.wpcs_version == 'dev-master' && ' + Lint' || '' }}: PHP ${{ matrix.php }} - PHPCS ${{ matrix.phpcs_version }} - WPCS ${{ matrix.wpcs_version }}"
```

Comment:
Same as before, could `php` be `php_version`?

Reply:
If a `lint` variable is added to the matrix, this condition would be a lot smaller and simpler.

Reply:
True, but it would mean you'd have to add the `lint` key to every build in the matrix and make sure not to forget to toggle it to the right value and keep it in sync with the condition now used, when adding new builds, while now things are handled automatically.

In other words: yes, the condition would be cleaner, but the matrix would not be and the script would become more prone to human error if someone less versed in GH Actions would edit the script.

Reply:
Note: this may not seem like a big thing for the limited matrix currently used by this repo, but the matrices for PHPCS repos can get quite complicated - see https://github.com/PHPCSStandards/PHPCSUtils/blob/develop/.github/workflows/test.yml#L41-L124 for an example.

Reply:
Another remark _against_ adding a lint key: it would also force us to pull the matrix in the `test` script apart.

The matrix in the test script currently looks like this:
```yml
        php_version: ['5.4', '5.5', '5.6', '7.0', '7.1', '7.2', '7.3', '7.4', '8.0']
        phpcs_version: ['3.6.0', 'dev-master']
        wpcs_version: ['2.3.0', 'dev-master']
        experimental: [false]
```

This is automatically expanded to give use four builds against every PHP version.

If I were to add a `lint` key, the default would have to be `false` as we only want it to be applied to the `dev-master` (PHPCS + WPCS) builds so it will only run once for every PHP version.

In effect this means that I'd have to add the complete set of builds to which that key should be applied (`true`) in the `include` section making the matrix much more wordy and less intuitive to read and manage as I'd be changing builds added in the matrix via `include`, instead of adding new builds via `include` (both are possible in GH Actions).

In that case, what I'd much rather propose would be to remove the linting from this build and add a separate script just to do the linting.